test/e2e/scheduling/priorities.go

```diff
@@ -567,10 +570,14 @@ func createBalancedPodForNodes(f *framework.Framework, cs clientset.Interface, n
 			},
 		}
 
-		err := testutils.StartPods(cs, 1, ns, string(uuid.NewUUID()),
-			*initPausePod(f, *podConfig), true, framework.Logf)
+		go func(){
+			err := testutils.StartPods(cs, 1, ns, string(uuid.NewUUID()),
+				*initPausePod(f, *podConfig), true, framework.Logf)
+			errChan <- err
+		}()
 
-		if err != nil {
+		select {
+		case err := <-errChan:
```

Comment:
This doesn't look correct. This will be satisfied when any `StartPods` function finishes, with or without an error.

We better use a wait group.

BTW, could there be a node or a set of nodes where we don't need to create any pod?

Reply:
Good point, I updated this to use a waitgroup. We do still need to handle the errors if necessary, so I changed the `select` statement to add a conditional check for `err != nil`.

To answer your question, we are creating a pod on every node that's at least the cri-o minMemLimit (from https://github.com/kubernetes/kubernetes/pull/98073)

Reply:
Or rather, a node that already has the desired cpu/memory ratio.

Reply:
Nope, doing so could cause the problem described in https://github.com/kubernetes/kubernetes/pull/98073. For example, if one node already has the target ratio and another node is just below it, the requested balancing resources could be less than the cri-o minimum. If every node gets at least 1 pod that is requesting at least the cri-o minimum, then we know that all balancing pods will request sufficient memory.